src/Microsoft.AspNetCore.Server.Kestrel/KestrelServerLimits.cs

```diff
@@ -21,6 +24,30 @@ public class KestrelServerLimits
         private int _maxRequestHeaderCount = 100;
 
         /// <summary>
+        /// Gets or sets the maximum size of the response buffer before calls
+        /// write calls begin to block or return incomplete tasks.
+        /// </summary>
+        /// <remarks>
+        /// When set to null, the size of the response buffer is unlimited.
+        /// Defaults to 65536 bytes (64 KB).
+        /// </remarks>
+        public long? MaxResponseBufferSize
```

Comment:
Sure, that has the same problem as this. When if you end up in a situation where the consumer is reading slowly, you'll end up pausing and resuming constantly. Maybe we can add it later...